From 281616b481b48ab04d211c37ef20261444567dfa Mon Sep 17 00:00:00 2001
From: lei zhilin <leizhilin@eswincomputing.com>
Date: Tue, 13 May 2025 14:37:57 +0800
Subject: [PATCH 359/540] Revert "<fix>(2d):hae power off delay by timeout"

This reverts commit ba8cd3fad9fbee53ef7d7f3f02e1d051f0e73ee4.

Reason for revert: <INSERT REASONING HERE> smoke failed
---
 .../hal/kernel/arch/gc_hal_kernel_hardware.c  | 31 ++++++-------------
 .../eswin/hae/hal/kernel/inc/gc_hal_base.h    |  3 --
 .../os/linux/kernel/gc_hal_kernel_device.c    |  3 +-
 .../os/linux/kernel/gc_hal_kernel_parameter.h |  2 +-
 4 files changed, 11 insertions(+), 28 deletions(-)

diff --git a/drivers/staging/media/eswin/hae/hal/kernel/arch/gc_hal_kernel_hardware.c b/drivers/staging/media/eswin/hae/hal/kernel/arch/gc_hal_kernel_hardware.c
index 1eaf7e4ebdf3b..0cd8c0ab4daed 100644
--- a/drivers/staging/media/eswin/hae/hal/kernel/arch/gc_hal_kernel_hardware.c
+++ b/drivers/staging/media/eswin/hae/hal/kernel/arch/gc_hal_kernel_hardware.c
@@ -5139,19 +5139,6 @@ gckHARDWARE_PowerControlClusters(gckHARDWARE Hardware,
 
     command = Hardware->kernel->command;
 
-    do {
-        gckOS_Udelay(Hardware->os, 1000);
-
-        gcmkONERROR(gckOS_ReadRegisterEx(Hardware->os, Hardware->kernel,
-                                            0x00004, &idle));
-
-        timer++;
-        if (timer >= 10) {
-            hae_print("wait register idl: 0x%x timeout!", idle);
-            break;
-        }
-    } while (!_IsHWIdle(idle, Hardware));
-
 #if !gcdFPGA_BUILD
     if (Hardware->identity.customerID != 0x85 || !Hardware->mcFE ||
         Hardware->options.enableNNClusters == (gctUINT32)~0UL ||
@@ -6135,15 +6122,6 @@ gckHARDWARE_SetPowerState(gckHARDWARE Hardware, gceCHIPPOWERSTATE State)
     }
 
 #if gcdPOWEROFF_TIMEOUT
-    if (state == gcvPOWER_ON) {
-        Hardware->nextPowerState = gcvPOWER_INVALID;
-    }
-
-    if (!timeout && state == gcvPOWER_OFF) {
-        status = gcvSTATUS_OK;
-        goto OnError;
-    }
-
     if (timeout && Hardware->nextPowerState == gcvPOWER_INVALID) {
         /* Delayed power state change is canceled. */
         status = gcvSTATUS_OK;
@@ -6297,6 +6275,15 @@ gckHARDWARE_SetPowerState(gckHARDWARE Hardware, gceCHIPPOWERSTATE State)
 #endif
 
 #if gcdPOWEROFF_TIMEOUT
+    if (!broadcast) {
+        /*
+         * Cancel delayed power state change.
+         * Stop timer is not as good as set as no state change. Timer may run
+         * into this function already when try to stop the timer.
+         */
+        Hardware->nextPowerState = gcvPOWER_INVALID;
+    }
+
     if (Hardware->powerOffTimeout &&
         (state == gcvPOWER_IDLE || state == gcvPOWER_SUSPEND)) {
         /* Delayed power off. */
diff --git a/drivers/staging/media/eswin/hae/hal/kernel/inc/gc_hal_base.h b/drivers/staging/media/eswin/hae/hal/kernel/inc/gc_hal_base.h
index 986d2f0465274..2af1f18c77543 100644
--- a/drivers/staging/media/eswin/hae/hal/kernel/inc/gc_hal_base.h
+++ b/drivers/staging/media/eswin/hae/hal/kernel/inc/gc_hal_base.h
@@ -2906,9 +2906,6 @@ gcoOS_ProfileDB(IN gctCONST_STRING Function, IN OUT gctBOOL_PTR Initialized);
 #define gcmOPT_POINTER(ptr)            (((ptr) == gcvNULL) ? gcvNULL : *(ptr))
 #define gcmOPT_STRING(ptr)             (((ptr) == gcvNULL) ? "(nil)" : (ptr))
 
-#define hae_print(fmt, ...) \
-    printk("[%s:%d] "fmt"\n", __func__, __LINE__, ##__VA_ARGS__)
-
 void
 gckOS_Print(IN gctCONST_STRING Message, ...) CHECK_PRINTF_FORMAT(1, 2);
 
diff --git a/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_device.c b/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_device.c
index af46c7629dc85..9699a7b40d148 100644
--- a/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_device.c
+++ b/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_device.c
@@ -2268,8 +2268,7 @@ isrRoutine(int irq, void *ctxt)
     }
 
     if(status == gcvSTATUS_GENERIC_IO){
-        hae_print("irq: %d, hw: %p, addr: %llx clock close.\n",
-            irq, kernel->hardware, kernel->hardware->lastExecuteAddress);
+        pr_info("[%s] hae isr meet clock close...\n", __func__);
     }
 
     return IRQ_NONE;
diff --git a/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_parameter.h b/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_parameter.h
index 22f6ebe226a67..dcc6b0558a338 100644
--- a/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_parameter.h
+++ b/drivers/staging/media/eswin/hae/hal/os/linux/kernel/gc_hal_kernel_parameter.h
@@ -305,7 +305,7 @@ static int compression = -1;
 module_param(compression, int, 0644);
 MODULE_PARM_DESC(compression, "Disable compression if set it to 0, enabled by default");
 
-static int powerManagement = 1;
+static int powerManagement = 0;
 module_param(powerManagement, int, 0644);
 MODULE_PARM_DESC(powerManagement, "Disable auto power saving if set it to 0, enabled by default");
 
-- 
2.53.0

